From 0d9bb3c3d5ff1d706795f8b24362dc3de7945945 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Wed, 4 Jul 2012 17:32:13 +0800
Subject: ENGR00215762 OV5642 camera:Improve XGA@30 image quality

This patch changes output drive capability of ov5642
camera register 0x302c's bit[6:5] to 0'11 to get 4x
drive capability so that green lines cannot be seen
in the captured frames.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
(cherry picked from commit f3e5d9944e989e66167cbbaa4f3433d8a9baf84d)
---
 drivers/media/video/mxc/capture/ov5642.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/video/mxc/capture/ov5642.c b/drivers/media/video/mxc/capture/ov5642.c
index d5920b8..8d317fa 100644
--- a/drivers/media/video/mxc/capture/ov5642.c
+++ b/drivers/media/video/mxc/capture/ov5642.c
@@ -1188,7 +1188,7 @@ static struct reg_value ov5642_setting_30fps_XGA_1024_768[] = {
 	{0x5087, 0x00, 0, 0}, {0x5088, 0x00, 0, 0}, {0x5089, 0x00, 0, 0},
 	{0x302b, 0x00, 0, 0}, {0x3621, 0x87, 0, 0}, {0x3a00, 0x78, 0, 0},
 	{0x3808, 0x04, 0, 0}, {0x3809, 0x00, 0, 0}, {0x380a, 0x03, 0, 0},
-	{0x380b, 0x00, 0, 0}, {0x3815, 0x02, 0, 0},
+	{0x380b, 0x00, 0, 0}, {0x3815, 0x02, 0, 0}, {0x302c, 0x60, 0x60, 0},
 };
 
 static struct reg_value ov5642_setting_15fps_XGA_1024_768[] = {
-- 
1.7.9.5

